toml/decoder.py

```diff
@@ -957,3 +964,54 @@ def load_array(self, a):
                     atype = ntype
                 retval.append(nval)
         return retval
+
+    def handle_inline_comment(self, line, currentlevel):
+        return line
+
+    def handle_comment(self, currentlevel, comment_no, recent, line):
+        return comment_no
+
+    def comment_lint(self, sl, i, beginline):
+        j = i
+        try:
+            while sl[j] != '\n':
+                sl[j] = ' '
+                j += 1
+        except IndexError:
+            raise IndexError()
+        return beginline
+
+
+class TomlPreserveCommentDecoder(TomlDecoder):
+
+    def __init__(self, _dict=OrderedDict):
```

Comment:
Default `_dict` should not be `OrderedDict`

Reply:
Using ordered dict makes for faster dumping and also easier to code. 
I will create make provision when normal dict is used.

Reply:
Resolved